src/hotspot/share/opto/escape.cpp

```diff
@@ -1004,7 +1010,8 @@ void ConnectionGraph::reduce_phi_on_cmp(Node* cmp) {
     res_phi->set_req(i, res_phi_input);
   }
 
-  Node* new_cmp = _igvn->transform(new CmpINode(res_phi, zero));
+  // This CMP always compares whether the output of "res_phi" is TRUE as far as the "mask".
+  Node* new_cmp = _igvn->transform(new CmpINode(res_phi, (mask == BoolTest::mask::eq) ? one : zero));
```

Comment:
Why we do `res_phi` transform (line 988) when we create it instead of after we assign inputs (after loop at line 1012)?

Reply:
Probably just a copy and paste error.